src/snmp_agent.cc

```diff
@@ -339,11 +341,14 @@ snmp_prfSysFn(variable_list * Var, snint * ErrP)
                                       SMI_COUNTER32);
         break;
 
-    case PERF_SYS_MEMUSAGE:
+    case PERF_SYS_MEMUSAGE: {
+        Mem::PoolStats stats;
+        Mem::GlobalStats(stats);
         Answer = snmp_var_new_integer(Var->name, Var->name_length,
-                                      (snint) statMemoryAccounted() >> 10,
+                                      (snint) stats.meter->alloc.currentLevel() >> 10,
                                       ASN_INTEGER);
-        break;
+    }
+    break;
```

Comment:
The original location/indentation of the `break` statement looks more logical/natural to me:

```suggestion
        break;
    }
```

I do not insist on this change.

Reply:
The weird indentation is an astyle thing. I am reluctant to move the break statement into sub-scope as that implies situations where the switch-case falls through.

Reply:
> [moving] the break statement into sub-scope ... implies situations where the switch-case falls through.

There is no such implication IMO: The curly braces factually and visually clearly apply to the entire `case` statement (as they should).

Reply:
```c++
case W: {...}
case X:
 {...}
case Y:
  break;
```

Which **one** falls through and which **two** do not?

Reply:
```c++
case W: {...}
case X:
 {...}
case Y:
  break;
```
 
> Which **one** falls through and which **two** do not?

In a correctly structured code, like the one this change request is about, none fall through. Such code, including code this change request is about, marks fall through cases explicitly and does not position curly braces poorly (and inconsistently).

In some hypothetical poorly written code, like the one sketched in your example, it does not really matter which ones fall through. The code needs to be polished before it makes sense to discuss such style issues. We should not even discuss code like that in this change request.